modules/op-running-unpriveleged-builds-in-a-container.adoc

```diff
@@ -0,0 +1,45 @@
+[id='op-running-unpriveleged-builds-in-a-container._{context}']
+= Running unprivileged builds in a container
+
+You can run builds as a non-root user inside a container. To run as a non-root user inside the container, you need your container to do some set up to add the `subuid` and `subgid` with the latest shadow-utils.
+
+The public image based on Red Hat UBI has everything already set up for us in its https://catalog.redhat.com/software/containers/ubi8/buildah/602686f7b16b1eb2e30807ee?container-tabs=dockerfile[Containerfile]. As an extra step, you need to “force” running the container as the “build” user since if we leave it to the default, it will be “randomed” and would not have the setup needed to run as non-root.
+
+You can directly edit or export to a specific namespace the buildah ClusterTask as described in the previous section. You can add this `securityContext`` to each https://github.com/tektoncd/operator/blob/main/cmd/openshift/operator/kodata/tekton-addon/addons/02-clustertasks/buildah/buildah-task.yaml#L63[build] and https://github.com/tektoncd/operator/blob/main/cmd/openshift/operator/kodata/tekton-addon/addons/02-clustertasks/buildah/buildah-task.yaml#L75[push] steps.
+
+[source,yaml,subs="attributes+"]
+----
+securityContext:
+    runAsUser: 1000
+----
+
+Since we are building as user, buildah will keep its image caching in `/home/build/.local/cache`. Hence instead of volume mounting `/var/lib/` containers we are going to change the `mountPath` on build and push steps to the user one :
+
+[source,yaml,subs="attributes+"]
+----
+volumeMounts:
+    - name: varlibcontainers
+      mountPath: /home/build/.local/share/containers
+----
+
+You can test it using https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md#configmap[ConfigMap workspace] along with a sample DockerFile and a sample taskrun.
+
+[source,terminal,subs="attributes+"]
```

Comment:
This verification step should be split into two blocks: one with the command code block for `$ tkn tr logs -Lf buildah-run` and then a sample output section.